domain/src/main/scala/BitcoinCryptography.scala

```diff
@@ -1,13 +1,13 @@
 package org.parsec.protocol
 
-trait BitcoinCryptography {
-  // TODO: implement me
-  def bitcoinP2PKHAddressFromPublicKey(key: String): String = {
-    "btc-public-key"
-  }
+import org.apache.commons.codec.binary.Hex
+import org.bitcoinj.core.{ECKey, NetworkParameters}
 
-  // TODO: implement me
-  def bitcoinP2SHAddressFromPublicKey(key: String): String = {
-    "btc-public-key"
+trait BitcoinCryptography {
+  def bitcoinAddressFromPublicKey(key: String): String = {
```

Comment:
As far as I've been able to see. the `bitcoinj` library takes care of this complexity for us. Not 100% sure yet though but I would take it as a working assumption